src/mnt_setup_fnc.cpp

```diff
@@ -1,66 +1,101 @@
+
 #include <TChain.h>
 #include <TCanvas.h>
 #include <TH1F.h>
 #include <THStack.h>
 #include <string>
+#include <vector>
 #include <iostream>
 
+
 std::string const DATADIR = "/storage/epp2/phshgg/DVTuples__v23/";
 
 struct path_data {
-	std::string const DATADIR;
-	std::string sqrts,
+	std::string const DATADIR,
+	sqrts,
 	stripping,
 	year,
 	polarity,
 	stream;
 };
 
-void make_histogram(struct path_data path_in, std::string hist_name, std::string hist_setup, int const nbins, float const xmin, float const xmax, std::string expression, THStack *hist_stack) {
+struct plot_config {
+	std::string const expression;
+	int const nbins;
+	float const xmin,
+	xmax;
+	std::string const label,
+	unit;
+
+  //create constructor to allow struct to configure correctly in the emplace_back function
+  plot_config(std::string const expression, int const nbins, float const xmin, float const xmax, std::string const label, std::string const unit)
+    :expression(std::move(expression))
+    ,nbins(nbins)
+    ,xmin(xmin)
+    ,xmax(xmax)
+    ,label(std::move(label))
+    ,unit(std::move(unit))
+  {}
+};
+
+std::vector<plot_config> plot_configurations;
+
+
+TH1F* make_TH1F(struct path_data path_in, struct plot_config hist_input, std::string hist_name, std::string hist_text) {
 	TChain ch("Z/DecayTree");
 	std::string path = path_in.DATADIR + path_in.sqrts + "TeV_" + path_in.year + "_" + path_in.stripping + "_" + path_in.polarity + "_" + path_in.stream + ".root";
 	ch.Add(path.c_str());
 	
-	TH1F *hist = new TH1F(hist_name.c_str(), hist_setup.c_str(), nbins, xmin, xmax);
-	expression += hist->GetName();
+	TH1F *hist = new TH1F(hist_name.c_str(), hist_text.c_str(), hist_input.nbins, hist_input.xmin, hist_input.xmax);
+	std::string expression = hist_input.expression +">>"+ hist->GetName();
 	ch.Draw(expression.c_str());
-	hist_stack->Add(hist);
+	return hist;
 }
 
+void plot_data_sim(struct path_data mnt_in, struct path_data sim_in, struct plot_config hist_input) {
+	std::string hist_title = "Measurement vs Simulation for " + hist_input.label,
+	  x_axis = hist_input.label + " " + hist_input.unit,
+	  sim_text = hist_input.label + " Simulation Data;" + x_axis,
+	  mnt_text = hist_input.label + " Experimental Data;" + x_axis;
+
+	TH1F *hist_mnt = make_TH1F(mnt_in, hist_input, "measurement", mnt_text);

```

Comment:
To avoid the memory leak warning I think you need to choose a unique `name` for each histogram that you make with `* = new TH1...`. 
I think your lower-level plotting code already uses `hist->GetName()` so that the expression is drawn into the right histogram, so it shouldn't matter too much what name you give. Presumably do `"measurement"+hist_input.label` and it will just work.

Reply:
This actually resolved itself using the hist->SetTitle("") from the earlier comment.

Apologies for highlighting it again when I updated the code, your comment hadn't appeared for me yet!